src/main/java/gyro/azure/compute/VirtualMachineResource.java

```diff
@@ -629,206 +750,321 @@ public void create(GyroUI ui, State state) {
-                    .withCustomData(getEncodedCustomData())
-                    .withExistingDataDisk(client.disks().getById(getDisk().getId()));
-
-            } else if (getVmImageType().equals("stored")) {
-                createUnmanaged = withOS.withStoredWindowsImage(getStoredImage())
-                    .withAdminUsername(getAdminUserName()).withAdminPassword(getAdminPassword());
-
-            } else if (getVmImageType().equals("custom") || getVmImageType().equals("gallery")) {
-                WithWindowsAdminPasswordManaged passwordManaged;
-                if (getVmImageType().equals("custom")) {
-                    passwordManaged = withOS.withWindowsCustomImage(getCustomImage())
-                        .withAdminUsername(getAdminUserName());
-                } else {
-                    passwordManaged = withOS.withWindowsGalleryImageVersion(getGalleryImageVersion())
-                        .withAdminUsername(getAdminUserName());
-                }
-
-                createManaged = passwordManaged.withAdminPassword(getAdminPassword());
+    /**
+     * Third step in Virtual Machine Fluent workflow. Splits workflow path by Image Type (OS Disk)
+     * Configures OS Disk, Admin User, and Data Disks
+     * @return {@link WithCreate} VM Definition object ready for final generic configurations
+     */
+    private WithCreate configureLinux(Azure client, WithOS withOS) {
+        switch (getVmImageType()) {
+            case "custom":
+                return configureLinuxManaged(
+                        client,
+                        withOS.withLinuxCustomImage(getCustomImage()));
+            case "gallery":
+                return configureLinuxManaged(
+                        client,
+                        withOS.withLinuxGalleryImageVersion(getGalleryImageVersion()));
+            case "latest":
+                return configureLinuxManagedOrUnmanaged(
+                        client,
+                        withOS.withLatestLinuxImage(getImagePublisher(), getImageOffer(), getImageSku()));
+            case "popular":
+                return configureLinuxManagedOrUnmanaged(
+                        client,
+                        withOS.withPopularLinuxImage(KnownLinuxVirtualMachineImage.valueOf(getKnownVirtualImage())));
+            case "specific":
+                return configureLinuxManagedOrUnmanaged(
+                        client,
+                        withOS.withSpecificLinuxImageVersion(
+                                client.virtualMachineImages()
+                                .getImage(getImageRegion(), getImagePublisher(), getImageOffer(), getImageSku(), getImageVersion())
+                                .imageReference()));
+            case "stored":
+                return configureLinuxUnmanaged(
+                        client,
+                        withOS.withStoredLinuxImage(getStoredImage()));
+            case "specialized":
+                //TODO handle unmanaged OS disks?
```

Comment:
This case does not handle configuring data disks and so results in the following:

```
#vm-config-snip 
azure::virtual-machine virtual-machine-example
    name: "virtual-machine-example"
    resource-group: $(azure::resource-group resource-group-example-VM)
    network: $(azure::network network-example-VM)
    subnet: "subnet1"
    os-type: "linux"
    data-disks: [$(azure::disk disk-example-VM), $(azure::disk disk-example-VM-2)]
    os-disk: $(azure::disk disk-example-VM-OS)
    network-interface: $(azure::network-interface network-interface-example-VM)
    #private-ip-address: "10.0.0.10"
    #public-ip-address: $(azure::public-ip-address public-ip-address-example-VM)
    vm-image-type: "specialized"
    #known-virtual-image: "UBUNTU_SERVER_14_04_LTS"
    admin-user-name: "qwerty@123"
    admin-password: "qwerty@123"
    caching-type: "NONE"
    vm-size-type: "STANDARD_D1"
    storage-account-type-data-disk: "STANDARD_LRS"
    storage-account-type-os-disk: "STANDARD_LRS"
    custom-data: "custom data"
    tags: {
        Name: "network-interface-example-VM"
    }
    enable-system-managed-service-identity: false
    delete-os-disk-on-terminate: true
end
```

```
+ Creating azure::virtual-machine virtual-machine-example
· admin-password: 'qwerty@123'
· admin-user-name: 'qwerty@123'
· caching-type: 'NONE'
· custom-data: 'custom data'
· data-disks: [ azure::disk disk-example-VM-2, azure::disk disk-example-VM ]
· delete-os-disk-on-terminate: true
· enable-system-managed-service-identity: false
· name: 'virtual-machine-example'
· network: azure::network network-example-VM
· network-interface: azure::network-interface network-interface-example-VM
· os-disk: azure::disk disk-example-VM-OS
· os-type: 'linux'
· resource-group: azure::resource-group resource-group-example-VM
· storage-account-type-data-disk: 'STANDARD_LRS'
· storage-account-type-os-disk: 'STANDARD_LRS'
· subnet: 'subnet1'
· tags: { Name: 'network-interface-example-VM' }
· vm-image-type: 'specialized'
· vm-size-type: 'STANDARD_D1'
OK
```

```
$ ../../gyro/cli/dist/gyro up compute/virtual-machine.gyro --verbose
↓ Loading plugin: gyro:gyro-azure-provider:0.99.1-SNAPSHOT
⟳ Refreshed resources: 9

Looking for changes...

⟳ Update azure::virtual-machine virtual-machine-example (/subscriptions/137c2421-b110-4b7a-9c6b-a4b9cf0e7455/resourceGroups/resource-group-example-VM/providers/Microsoft.Compute/virtualMachines/virtual-machine-example)
· data-disks:  + azure::disk disk-example-VM-2, + azure::disk disk-example-VM
```